### DISCUSSION TITLE
FYI: Ran into some issues getting started without using the starter

### DISCUSSION THREAD

**Discussion by chrisjones314** [markdoc/markdoc#115] [Bug report] [answered]
I just wanted to let you know that I ran into some issues using the documentation to get started.   These issues may be due to my not understanding next.js/npm that well.   But, they might be things that other beginners would run into as well, which would prevent them from getting farther with markdoc.   Just FYI:

1.  I installed next.js and markdoc successfully, and got some markdown to render.
2. I wanted to get prism working, so I attempted to use the prism code from the 'common examples'.  I copied that code into a file under markdoc/nodes/fence.js.  
4. I had the impression that markdoc would somehow automatically load the schema under the markdoc folder, but it didn't seem to be loading anything.
5. I was surprised that in the example, the fence definition didn't seem to be exported.  I tried exporting it as default, like this:

```javascript
import 'prismjs';
import 'prismjs/themes/prism.css';

import Prism from 'react-prism';

export function Fence({ children, language }) {
  console.log("Ran fence!!");
  return (
    <Prism key={language} component="pre" className={`language-${language}`}>
      {children}
    </Prism>
  );
}

const fence = {
  render: 'Fence',
  attributes: {
    language: {
      type: String,
      description:
        'The programming language of the code block. Place it after the backticks.'
    }
  }
};

export default fence;
```

That didn't work.   I also tried adding a schemaPath of "./markdoc/" in next.config.js, and it still didn't seem to load.

6. Finally, I added a markdown/config.js.   After various errors, I ended up with this:

```javascript
//import tags from './tags';
import fence from './nodes/fence';
//import functions from './functions';

const nodes = [
  fence,
]

const config = {
  nodes,
  // add other stuff here
}

export default config;
```

Which did finally seem to load the fence definition.

6. I got an error on this line

`import Prism from 'react-prism';`

Saying react-prism didn't exist.   I did an npm install on react-prism, and got this:

```bash
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: afib_website_nextjs@0.1.0
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"18.2.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^15.5.0 || ^16.0.0" from react-prism@4.3.2
```

When I looked, it looked like react-prism hadn't been updated since 2018.    At this point, I considered giving up, because my biggest concern with javascript-based software is dependency issues.   However, I went back, and tried cloning your starter repository, and it worked just fine.   When I looked at the actual fence code in that repository, it is not the same as your documentation example:

```javascript
export const fence = {
  render: CodeBlock,
  attributes: {
    content: {type: String},
    language: {
      type: String,
      description:
        'The programming language of the code block. Place it after the backticks.',
    },
  },
};
```

It appears to use CodeBlock and doesn't reference Prism at all.  

I am not an expert in javascript/npm, so I may just misunderstand things, but I wonder if your existing documentation may also yield discouraging results for others trying to get started.  If I had one suggestion, it would be tell people flat out to use starter, rather than implying that they might want to start with "getting started with next.js"

Thanks for publishing markdoc.   I think it seems very attractive conceptually and I am looking forward to trying to work with it further.

**Comment by mfix-stripe** [comment, marked as answer]
@chrisjones314 apologies for these difficulties. Thanks for such a thorough right up above—I can see a number of issues in your examples for why this isn't working for you. 

For now though, instead of wiring up all of this yourself, do you want to try just cloning our starter repo (https://github.com/markdoc/markdoc-starter)? 🙂 

This repo comes pre-configured with Markdoc, Next.js, and Prism (as you can see [here](https://markdoc-starter-stripe-oss-stripe-dev.vercel.app/)).

**Reply by chrisjones314** [reply]
Yes, that is what I did.  It worked great, and I am moving forward with it.   I was just trying to give you a heads up about how a beginner might go wrong trying to set it up themselves. Thanks!

**Reply by mfix-stripe** [reply]
Thanks @chrisjones314! Sorry, since this discussion was in the "Bug Report" category I thought about it differently. Thanks for helping out our community!

**Comment by songz** [comment]
Unfortunately, `Next.js` was not an option for me. Took me a bit of exploring around but I got a working piece of code. Hopes this helps someone in the future!

The [existing doc for syntax highlighting](https://markdoc.io/docs/examples#syntax-highlighting) 100% does not work currently. I could send a PR to fix if you could point me to where it is. 

In `./src/CodeBlock.js`:
```jsx
import * as React from "react";
import Prism from "prismjs";

import "prismjs/themes/prism-okaidia.css";
import "prismjs/plugins/line-numbers/prism-line-numbers.js";
import "prismjs/plugins/line-numbers/prism-line-numbers.css";

export function CodeBlock({ children, language }) {
  const ref = React.useRef(null);

  React.useEffect(() => {
    if (ref.current) {
      Prism.highlightElement(ref.current, false);
    }
  }, [children]);

  return (
    <pre className={`language-${language} line-numbers`}>
      <code ref={ref}>{children}</code>
    </pre>
  );
}

export const fence = {
  render: "CodeBlock",
  attributes: {
    content: { type: String },
    language: {
      type: String,
      description:
        "The programming language of the code block. Place it after the backticks.",
    },
  },
};
```

In `./src/client.js` (entry point)

```jsx
import Markdoc from "@markdoc/markdoc";

import { CodeBlock, fence } from "./components/CodeBlock.js";


    const ast = Markdoc.parse(mdContent);
    const content = Markdoc.transform(ast, {
      variables: markdocVariables,
      nodes: {
        fence,
      },
    });
    const children = Markdoc.renderers.react(content, React, {
      components: {
        CodeBlock,
      },
    });

```
